src/pages-and-resources/discussions/DiscussionsSettings.jsx

```diff
@@ -24,80 +21,81 @@ import { fetchApps } from './data/thunks';
 import AppList from './app-list';
 import AppConfigForm from './app-config-form';
 
+const SELECTION_STEP = 'selection';
+const SETTINGS_STEP = 'settings';
+
 function DiscussionsSettings({ courseId, intl }) {
   const dispatch = useDispatch();
-  const { path } = useRouteMatch();
-  const { pathname } = useLocation();
   const { path: pagesAndResourcesPath } = useContext(PagesAndResourcesContext);
 
   useEffect(() => {
     dispatch(fetchApps(courseId));
   }, [courseId]);
 
   const discussionsPath = `${pagesAndResourcesPath}/discussions`;
-  const isFirstStep = pathname === discussionsPath;
+  const { params: { appId } } = useRouteMatch();
 
-  const steps = [{
-    label: intl.formatMessage(messages.providerSelection),
-    iconLabel: isFirstStep ? undefined : (
-      <Check style={{ width: '1rem', height: '1rem' }} />
-    ),
-    incomplete: false,
-  },
-  {
-    label: intl.formatMessage(messages.settings),
-    incomplete: isFirstStep,
-  }];
+  const startStep = appId ? SETTINGS_STEP : SELECTION_STEP;
+  const [currentStep, setCurrentStep] = useState(startStep);
+
+  useEffect(() => {
+    setCurrentStep(appId ? SETTINGS_STEP : SELECTION_STEP);
+  }, [appId]);
 
   const handleClose = useCallback(() => {
     history.push(pagesAndResourcesPath);
-  }, [courseId]);
+  }, [pagesAndResourcesPath]);
 
   const handleBack = useCallback(() => {
-    if (isFirstStep) {
-      history.push(pagesAndResourcesPath);
-    } else {
```

Comment:
Why do we not need to worry about `SETTINGS_STEP` and `SELECTION_STEP` here while setting the history?  we are only setting `discussionsPath` in either case.

Reply:
Not sure I understand - the back button only appears on the second step of the stepper, in which case it always means to go back to the first step.  We do this by updating the URL and allowing the component to update its state in response.  i.e., it'll the component will decide that it's back on step one when it sees a URL without an `appId1